### PR TITLE
Trying to invite alias with long string causes not handled exception #864

### PR DESCRIPTION
**Pull request by jhyle**
added test for invite of too long alias and corrected TkInvite::act

### CODE REVIEW

**Comment by karato** [comment]
@jhyle Let me find a reviewer for this pull request, thanks for submitting it

**Comment by karato** [comment]
@carlosmiranda review it

File: netbout-web/src/main/java/com/netbout/rest/bout/TkInvite.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
  * Invite a friend to the bout.
  *
  * @author Yegor Bugayenko (yegor@teamed.io)
- * @version $Id$
+ * @version $Id: 9ee51b1fdc2de87f03cebf45e3c068b991a95432 $
```
Comment (carlosmiranda) [inline netbout-web/src/main/java/com/netbout/rest/bout/TkInvite.java]: @jhyle Don't change this line

File: netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java [hunk 2]
```diff
@@ -43,8 +44,10 @@
 /**
  * Test case for {@link TkInvite}.
  * @author Endrigo Antonini (teamed@endrigo.com.br)
- * @version $Id$
+ * @version $Id: 1cc7dffb02a2cc1e4f71ee5953e0ccc757dbe9d1 $
```
Comment (carlosmiranda) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java]: @jhyle Same here, don't change it please

File: netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java [hunk 3]
```diff
@@ -88,4 +91,46 @@ public void invitesUserByEmail() throws Exception {
...
+    /**
+     * TkInvite should throw an RsFailure when invite alias is too long.
+     * @throws Exception If there is some problem inside
+     */
+    @Test (expected = RsFailure.class)
+    public void inviteWithTooLongAlias() throws Exception {
+        final MkBase base = new MkBase();
+        final String urn = "urn:test:1";
+        final User user = base.user(new URN(urn));
+        final String name = "12345678901234567890123456789012345678901234"
```
Comment (carlosmiranda) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java]: @jhyle String concat is a bad idea, see [this article](http://www.yegor256.com/2014/06/19/avoid-string-concatenation.html) for alternatives.

File: netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java [hunk 4]
```diff
@@ -88,4 +91,46 @@ public void invitesUserByEmail() throws Exception {
             throw ex;
         }
     }
+
+    /**
+     * TkInvite should throw an RsFailure when invite alias is too long.
+     * @throws Exception If there is some problem inside
+     */
+    @Test (expected = RsFailure.class)
+    public void inviteWithTooLongAlias() throws Exception {
```
Comment (carlosmiranda) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java]: @jhyle Test name does not describe the scenario at all. Let's call it `failsWhenInvitingWithTooLongAlias`

File: netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java [hunk 5]
```diff
@@ -88,4 +91,46 @@ public void invitesUserByEmail() throws Exception {
...
+                new RqWithHeader(
+                    new RqWithAuth(
+                        urn,
+                        new RqFake(
+                            RqMethod.POST,
+                            String.format(
+                                "/b/%d/invite",
+                                bout.number()
+                            ),
+                            "name=" + name
```
Comment (carlosmiranda) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java]: @jhyle Same here, let's avoid string concat. In this case, you can use `String.format("name=%s", name)`

**Comment by carlosmiranda** [comment]
@jhyle Pretty good branch, please see my comments above.

**Comment by jhyle** [comment]
@carlosmiranda Thank you for review. The issues should be fixed now.

File: netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java [hunk 6]
```diff
@@ -45,6 +47,8 @@
  * @author Endrigo Antonini (teamed@endrigo.com.br)
  * @version $Id$
  * @since 2.15.1
+ * @checkstyle MultipleStringLiteralsCheck (500 lines)
+ * @checkstyle StringLiteralsConcatenationCheck (500 lines)
```
Comment (carlosmiranda) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkInviteTest.java]: @jhyle One more comment, let's remove this suppression.

**Comment by carlosmiranda** [comment]
@jhyle one more comment above

**Comment by jhyle** [comment]
@carlosmiranda done

**Comment by carlosmiranda** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@carlosmiranda 
 According to our [quality rules](http://www.teamed.io/qa.html):
 > Pull request description explains the solution proposed and contains a link to the original ticket it is related to.
 
 There is no link to the original ticket in description.
As far as you are the performer of this ticket you are responsible for its quality. Please, ask to fix this issue and make sure that everyone follows this rule in your tickets in the future.

**Comment by jhyle** [comment]
@elenavolokhova @carlosmiranda I added the ticket number.

**Comment by elenavolokhova** [comment]
@jhyle Thank you!

**Comment by elenavolokhova** [comment]
@karato Quality is acceptable here.

**Comment by karato** [comment]
> @karato Quality is acceptable here.

@elenavolokhova thanks for the QA review, we'll work better next time

**Comment by karato** [comment]
@carlosmiranda I added 10 mins to @elenavolokhova (for QA review) in transaction 77011892... thanks, paid, **24 mins** to your account, payment ID is `77011901`, task took 89 hours and 44 mins to complete... extra minutes for review comments (c=9)... +24 added to your rating, current score is: [+2864](http://www.netbout.com/b/35023?open=rating)

**Pull request merged**